flow/Arena.h

```diff
@@ -633,6 +633,9 @@ class StringRef {
 		return tokens;
 	}
 
+	// True if both StringRefs reference exactly the same memory
+	bool same(const StringRef& s) const { return data == s.data && length == s.length; }
```

Comment:
I had the same thought and also wasn't sure if memcmp does this.  It will almost never be true in normal StringRef use cases, so even though it's just a few instructions I don't think it's worth doing.

Reply:
I think there also is some clarity value in being explicit about the possibility that the two strings could be the same memory at the call sites for same().